core/src/main/java/io/seata/core/rpc/netty/AbstractRpcRemoting.java

```diff
@@ -230,24 +232,50 @@ private Object sendAsyncRequest(String address, Channel channel, Object msg, lon
         futures.put(rpcMessage.getId(), messageFuture);
 
         if (address != null) {
-            ConcurrentHashMap<String, BlockingQueue<RpcMessage>> map = basketMap;
-            BlockingQueue<RpcMessage> basket = map.get(address);
-            if (basket == null) {
-                map.putIfAbsent(address, new LinkedBlockingQueue<>());
-                basket = map.get(address);
-            }
-            basket.offer(rpcMessage);
-            if (LOGGER.isDebugEnabled()) {
-                LOGGER.debug("offer message: " + rpcMessage.getBody());
-            }
-            if (!isSending) {
-                synchronized (mergeLock) {
-                    mergeLock.notifyAll();
+            /*
+            The batch send.
+            Object From big to small: RpcMessage -> MergedWarpMessage -> AbstractMessage
+            @see AbstractRpcRemotingClient.MergedSendRunnable
+            */
+            if (NettyClientConfig.isEnableClientBatchSendRequest()) {
+                ConcurrentHashMap<String, BlockingQueue<RpcMessage>> map = basketMap;
+                BlockingQueue<RpcMessage> basket = map.get(address);
+                if (basket == null) {
+                    map.putIfAbsent(address, new LinkedBlockingQueue<>());
+                    basket = map.get(address);
+                }
+                basket.offer(rpcMessage);
+                if (LOGGER.isDebugEnabled()) {
+                    LOGGER.debug("offer message: {}", rpcMessage.getBody());
+                }
+                if (!isSending) {
+                    synchronized (mergeLock) {
+                        mergeLock.notifyAll();
+                    }
+                }
+            } else {
+                // the single send.
+                MergedWarpMessage mergeMessage = new MergedWarpMessage();
+                mergeMessage.msgs.add((AbstractMessage) rpcMessage.getBody());
+                mergeMessage.msgIds.add(rpcMessage.getId());
+                try {
+                    sendRequest(channel, mergeMessage);
```

Comment:
It ’s not the same as what I understand. The single piece I understand is to send RpcMessage directly.

Reply:
> It ’s not the same as what I understand. The single piece I understand is to send RpcMessage directly.

ok.